server.go

```diff
@@ -486,49 +499,83 @@ func NewFromConfig(config ServerConfig) (*Server, error) {
 	// initialize workers with state from *Server.IsWorker.
 	ret.ForwardAddr = conf.ForwardAddress
 
-	// Control whether Veneur should emit metric
-	// "veneur.flush.unique_timeseries_total", which may come at a
-	// slight performance hit to workers.
-	ret.CountUniqueTimeseries = conf.CountUniqueTimeseries
+	ret.enableMetricRouting = conf.Features.EnableMetricRouting
+	ret.computationRoutingConfig = conf.MetricComputationRouting
+	ret.WorkerSets = make([]WorkerSet, 0)
+	ret.subscribedFlushGroupsBySink = make(map[string][]string)
+	if ret.enableMetricRouting {
+		for _, config := range ret.computationRoutingConfig {
+			if config.WorkerCount < 1 {
+				logger.WithFields(logrus.Fields{
+					"worker_name":  config.Name,
+					"worker_count": config.WorkerCount,
+				}).Fatal("WorkerCount must be greater than 0")
+			}
+			logger.WithField("worker_name", config.Name).Info("Preparing WorkerSet")
+			workerSet := WorkerSet{make([]*Worker, config.WorkerCount), &config}
+			for i := 0; i < config.WorkerCount; i++ {
+				// TODO: ids are not unique like this?
+				workerSet.Workers[i] = NewWorker(i+1, ret.IsLocal(), ret.TraceClient, log, ret.Statsd)
+			}
+			ret.WorkerSets = append(ret.WorkerSets, workerSet)
+		}
+	} else {
+		numWorkers := 1
+		if conf.NumWorkers > 1 {
+			numWorkers = conf.NumWorkers
+		}
+		logger.WithField("number", numWorkers).Info("Preparing workers")
+		workers := make([]*Worker, numWorkers)
+		for i := 0; i < numWorkers; i++ {
+			workers[i] = NewWorker(i+1, ret.IsLocal(), ret.TraceClient, log, ret.Statsd)
+		}
+		ret.WorkerSets = append(ret.WorkerSets, WorkerSet{
+			workers,
+			&ComputationRoutingConfig{
+				"",
+				"",
+				MatcherConfigs{},
+				ret.interval,
+				0,
+				numWorkers,
+				true,
+			},
+		})
+	}
 
-	// Use the pre-allocated Workers slice to know how many to start.
-	for i := range ret.Workers {
-		ret.Workers[i] = NewWorker(i+1, ret.IsLocal(), ret.CountUniqueTimeseries, ret.TraceClient, log, ret.Statsd)
-		// do not close over loop index
-		go func(w *Worker) {
-			defer func() {
-				ConsumePanic(ret.TraceClient, ret.Hostname, recover())
-			}()
-			w.Work()
-		}(ret.Workers[i])
+	for _, workerSet := range ret.WorkerSets {
+		for _, worker := range workerSet.Workers {
+			go func(w *Worker) {
+				defer func() {
+					ConsumePanic(ret.TraceClient, ret.Hostname, recover())
+				}()
+				w.Work()
+			}(worker)
+		}
 	}
 
 	ret.EventWorker = NewEventWorker(ret.TraceClient, ret.Statsd)
 
-	// Set up a span sink that extracts metrics from SSF spans and
-	// reports them via the metric workers:
-	processors := make([]ssfmetrics.Processor, len(ret.Workers))
-	for i, w := range ret.Workers {
-		processors[i] = w
-	}
-	metricSink, err := ssfmetrics.NewMetricExtractionSink(processors, conf.IndicatorSpanTimerName, conf.ObjectiveSpanTimerName, ret.TraceClient, log, &ret.parser)
-	if err != nil {
-		return ret, err
+	for _, workerSet := range ret.WorkerSets {
```

Comment:
Need to think through the performance implications of just creating N == # of WorkerSet ssfmetrics sinks rather than one sink that is WorkerSet aware. But it was easier to implement this way.